glib/src/main_context_channel.rs

```diff
@@ -249,13 +249,41 @@ unsafe extern "C" fn dispatch<T, F: FnMut(T) -> Continue + 'static>(
     ffi::G_SOURCE_CONTINUE
 }
 
+#[cfg(feature = "v2_64")]
+unsafe extern "C" fn dispose<T, F: FnMut(T) -> Continue + 'static>(source: *mut ffi::GSource) {
+    let source = &mut *(source as *mut ChannelSource<T, F>);
+
+    if let Some(ref channel) = source.channel {
+        // Set the source inside the channel to None so that all senders know that there
+        // is no receiver left and wake up the condition variable if any
+        let mut inner = (channel.0).0.lock().unwrap();
+        inner.source = ChannelSourceState::Destroyed;
+        if let Some(ChannelBound { ref cond, .. }) = (channel.0).1 {
+            cond.notify_all();
+        }
+    }
+}
+
 unsafe extern "C" fn finalize<T, F: FnMut(T) -> Continue + 'static>(source: *mut ffi::GSource) {
     let source = &mut *(source as *mut ChannelSource<T, F>);
 
     // Drop all memory we own by taking it out of the Options
-    let channel = source.channel.take().expect("Receiver without channel");
 
+    #[cfg(feature = "v2_64")]
+    {
+        let _ = source.channel.take().expect("Receiver without channel");
+    }
+    #[cfg(not(feature = "v2_64"))]
     {
+        let channel = source.channel.take().expect("Receiver without channel");
+
+        // FIXME: This is the same as would otherwise be done in the dispose() function but
```

Comment:
Does marking this as `FIXME` make sense? It is not going to be fixed... that is why in new glib we have the proper fix :)

Reply:
Yes, when we have newer glib as minimum version we can fix it be removing the old code :) That was my thought here.